apps/app/src/mocks/miragejs/orm/factory/BaseFactory.ts

```
@@ -0,0 +1,206 @@
+        // Create a proxy to track property access
+        const proxy = new Proxy(attrs, {
+          get(target, prop) {
+            if (typeof prop === 'string' && prop in target) {
+              detectCycle(prop);
+            }
+            return target[prop as keyof typeof target];
+          },
+        });
+
+        // Call the function with the proxy as this context
+        (value as Function).call(proxy, modelId);
```

ESLint

[error] The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type.

(@typescript-eslint/no-unsafe-function-type)